src/optimistic-oracle/optimistic-oracle.js

```diff
@@ -0,0 +1,232 @@
+    throw new Error(`Error getting response from CoinGecko (status=${response.status})`);
+  }
+
+  // check that we got data for this token address
+  if (Object.keys(response.data).length === 0) {
+    throw new Error(`CoinGecko returned no data for token '${contractAddress}'`);
+  }
+
+  // in the response data, the token address key is NOT checksum encoded
+  const tokenAddress = contractAddress.toLowerCase();
+
+  // check that the currency denomination we requested is supported
+  if (!(Object.prototype.hasOwnProperty.call(response.data[tokenAddress], vsCurrency))) {
+    throw new Error(`CoinGecko has no data for token '${tokenAddress}' vs currency '${vsCurrency}'`);
+  }
+
+  const price = response.data[tokenAddress][vsCurrency];
+
+  return price;
+}
+
+function provideHandleTransaction(erc20Contract) {
+  return async function handleTransaction(txEvent) {
+    const findings = [];
+
+    if (optimisticOracleAddress === undefined) {
+      // get the Optimistic Oracle contract address for mainnet
+      // the address returned by the promise will be lowercase
+      optimisticOracleAddress = await getAddress('OptimisticOracle', CHAIN_ID);
+    }
+
+    // filter only logs that match the optimistic oracle address
+    // test transaction for ProposePrice event:
+    // npx forta-agent run --tx 0x76ff352b2665886a2a3d3b16fe0fa41f61e4ffc0824b3a6734383d397187f53f
+    const oracleLogs = txEvent.logs.filter((log) => log.address === optimisticOracleAddress);
+
+    if (oracleLogs === []) return findings;
+
+    // parse oracle logs for our target events:  RequestPrice and ProposePrice
+    const parse = (log) => iface.parseLog(log);
+    const filter = (log) => eventNames.indexOf(log.name) !== -1;
+    const parsedLogs = oracleLogs.map(parse).filter(filter);
+
+    // process the target events
+    /* eslint-disable no-await-in-loop */
+    for (let i = 0; i < parsedLogs.length; i++) {
+      const log = parsedLogs[i];
+      if (log.name === 'RequestPrice') {
+        // get token address (this will be checksum encoded)
+        const { currency, requester } = log.args;
+
+        // make a request to the price feed API
+        let price;
+        try {
+          price = await getPrice(currency);
+        } catch (error) {
+          console.error(error);
+          continue;
+        }
+
+        // report the price obtained as a finding
+        findings.push(Finding.fromObject({
+          name: 'UMA Price Request',
+          description: `Price requested from Optimistic Oracle.  Currency=${currency}, Price=${price}`,
+          alertId: 'AE-UMA-OO-REQUESTPRICE',
+          severity: FindingSeverity.Low,
+          type: FindingType.Unknown,
+          protocol: 'uma',
+          everestId: config.umaEverestId,
+          metadata: {
+            requester,
+            currency,
+            price,
+          },
+        }));
+      }
+
+      if (log.name === 'ProposePrice') {
+        const { currency } = log.args;
```

Comment:
I'm not certain why the destructuring assignment was used in the `Request Price` case but not here:
```suggestion
const { currency, requester, proposer } = log.args;
```

Reply:
Fixed